src/main/java/duke/Storage.java

```diff
@@ -0,0 +1,173 @@
+                break;
+            default:
+                System.out.println("Unable to determine type of task");
+                break;
+            }
+        }
+
+        return taskList;
+    }
+
+    private String formatTaskDateTime(String dateTime) {
+        assert dateTime.length() != 0 : "Date and time cannot be empty";
+        String[] dateTimes = dateTime.split(",");
+        String date = dateTimes[0]; // MMM DD YYYY
+        String time = dateTimes[1]; // HH:MM:SS
+
+        DateTimeFormatter format = DateTimeFormatter.ofPattern("MMM d yyyy");
+        LocalDate newDate = LocalDate.parse(date, format);
+        int monthInt = newDate.getMonthValue();
+        String year = String.valueOf(newDate.getYear());
+        int dayInt = newDate.getDayOfMonth();
+
+        String month = monthInt >= 10
+                ? String.valueOf(monthInt)
+                : "0".concat(String.valueOf(monthInt));
+
+        String day = dayInt >= 10
+                ? String.valueOf(dayInt)
+                : "0".concat(String.valueOf(dayInt));
+
+        String hourAndMinutes = time.substring(0, time.lastIndexOf(":"));
+
+        // Return YYYY/MM/DD HH:MM
+        String ans = year + "/" + month + "/" + day + hourAndMinutes;
+        return ans;
+    }
+
+    /**
+     * Saves all the Tasks from the TaskList into the user's local storage.
+     */
+    public void writeToStorage() {
+        try {
+            assert pathToStorage != null : "Path to storage cannot be empty";
+            File storageFile = new File(String.valueOf(pathToStorage));
+            FileWriter fw = new FileWriter(storageFile);
+            for (String task : this.allTasks) {
+                fw.write(task + '\n');
+            }
+            fw.close();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+        }
+    }
+
+    /**
+     * Adds a new Task to the local storage.
+     *
+     * @param task the Task to be added.
+     */
+    public void createTask(Task task) {
+        assert task != null : "Task cannot be null";
+        this.allTasks.add(task.toString());
+        writeToStorage();
+    }
+
+    /**
+     * Updates an existing task in the local storage.
+     *
+     * @param task the task to be updated.
+     * @param taskIndex the index of the Task in the List of all the Tasks.
+     */
+    public void updateTask(Task task, int taskIndex) {
+        assert task != null : "Task cannot be null";
+        assert taskIndex < allTasks.size() : "Task index must be within range of allTask size";
+        this.allTasks.set(taskIndex, task.toString());
+        writeToStorage();
+    }
+
+    /**
+     * Removes an existing from the local storage.
+     *
+     * @param taskIndex the index of the task in the List of all the Tasks to be deleted.
+     */
+    public void deleteTask(int taskIndex) {
+        assert taskIndex < allTasks.size() && taskIndex >= 0 : "Task index must be within range of allTask size";
```

Comment:
To improve the code quality, perhaps a boolean variable can be used to store allTasks.size() && taskIndex >= 0 here.